pkg/proxy/endpoint.go

```diff
@@ -154,3 +170,29 @@ func convertToMap(events []string) map[string]bool {
 
 	return eventsMap
 }
+
+func convertToMapAndSanitize(headers []string) map[string]string {
+	reg, err := regexp.Compile("[\x00-\x1f]+")
+
+	headerMap := make(map[string]string)
+
+	if err != nil {
+		// what to do if regex fails to compile?
+		fmt.Println("error with regex")
+		return headerMap
+	}
+
+	for _, header := range headers {
+
+		header = reg.ReplaceAllString(header, "")
+
+		splitHeader := strings.Split(header, ":")
```

Comment:
```suggestion
		splitHeader := strings.SplitN(header, ":", 2)
```

This way, the value can contain `:` characters.

Reply:
Good idea!